tests/macros/test-draft.js

```diff
@@ -0,0 +1,102 @@
+/* jshint node: true, mocha: true, esversion: 6 */
+
+var utils = require('./utils'),
+    chai = require('chai'),
+    chaiAsPromised = require('chai-as-promised'),
+    assert = chai.assert,
+    itMacro = utils.itMacro,
+    describeMacro = utils.describeMacro;
+
+// Let's add "eventually" to assert so we can work with promises.
+chai.use(chaiAsPromised);
+
+describeMacro('draft', function () {
+    itMacro('No arguments (en-US)', function (macro) {
+        return assert.eventually.equal(
+            macro.call(),
+            `<div class="overheadIndicator draft">\n    <p><strong>Draft</strong><br/>\n    This page is not complete.</p>\n    \n</div>`
+        );
+    });
+    itMacro('No arguments (es)', function (macro) {
+        macro.ctx.env.locale = 'es';
+        return assert.eventually.equal(
+            macro.call(),
+            `<div class="overheadIndicator draft">\n    <p><strong>Borrador</strong><br/>\n    Esta página no está completa.</p>\n    \n</div>`
+        );
```

Comment:
I totally agree with your concerns. We can easily switch to a slightly different style (where we add a handler to the promise returned from ``macro.call`` that does various checks on the actual result) if we'd rather check for specific elements in the HTML rather than do a simple exact string match.